cpp/src/c_api/random_walks.cpp

```diff
@@ -228,7 +229,7 @@ struct uniform_random_walks_functor : public cugraph::c_api::abstract_functor {
         (edge_weights != nullptr) ? std::make_optional(edge_weights->view()) : std::nullopt,
         raft::device_span<vertex_t const>{start_vertices.data(), start_vertices.size()},
         max_length_,
-        seed_);
+        rng_state_->rng_state_);
```

Comment:
This will fail in the C API.  You set `rng_state_` above to nullptr, so you're dereferencing a nullptr.

Until we add the rng_state as a parameter, I would suggest that you create a `raft::random::RngState{0}` either inline here or as a local variable above this call and pass it here.